tests/sidetrail/DEBUGGING.md

```diff
@@ -0,0 +1,140 @@
+# Debugging SideTrail
+
+This is a guide for what to do when SideTrail reports a potential timing violation / throws a red `x` in CI.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [How to locally rerun the tests](#how-to-locally-rerun-the-tests)
+- [Updating the patch files](#updating-the-patch-files)
+- [Undefined functions](#undefined-functions)
+- [Loop related errors](#loop-related-errors)
+- [Warning: No entrypoints found.](#warning-no-entrypoints-found)
+- [SideTrail fails with a long error trace](#sidetrail-fails-with-a-long-error-trace)
+- [What to do if the proof gets really slow](#what-to-do-if-the-proof-gets-really-slow)
+- [Expected runtimes:](#expected-runtimes)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## How to locally rerun the tests
+The first step to debugging is to locally rerun the failing proof(s).
+Follow [the instructions here](README.md#how-to-execute-the-tests) to get SideTrail docker container and execute the tests.
+
+## Updating the patch files
+
+The most common error you are likely to see if a failure to apply the necessary patches.
+Sidetrail depends on a set of patch files, stored in the `s2n/tests/sidetrail/working/patches`.
+These files patch away certain C constructs that SideTrail has trouble with.
+They are applied by the `copy_as_needed.sh` script in each proof.
+If the file they are patching is modified, `patch` may be unable to apply the needed patches, and fail with an error.
+
+```
+patching file utils/s2n_safety.c
+Hunk #1 FAILED at 57.
+1 out of 2 hunks FAILED -- saving rejects to file utils/s2n_safety.c.rej
+```
+
+To fix this, regenerate the patch file.
+
+1. View the failing patch to see what it does.
+   It may be useful to apply it to a previous version of the file, so you can see what the patched file looks like.
+   Identify any failing hunks, and figure out why they are failing to apply.
+   
+2. In a clean branch of s2n, modify the file in need of patching to have the required changes.
+   Work directly on the file in the s2n src folders - we will revery it back before committing our changes.
+   I often find it useful to also (temporarily) modify the `copy_as_needed.sh` file to just copy the file, and not patch it.
+
+3. Iterate on that file until the tests pass.
```

Comment:
We could be more specific and replace `that file` with `file in need of patching` to remove any confusion.